e2e-app/src/app/datepicker/datepicker.po.ts

```diff
@@ -1,7 +1,7 @@
 import {$} from 'protractor';
 
 export class DatepickerPage {
-  getDatepicker(selector = 'ngb-datepicker') { return $(selector); }
+  getDatepicker(selector = 'ngb-datepicker') { return $(selector) as any; }
```

Comment:
As for the other `ElementFinder as any`, this is needed due to additional type checks added for the updated Typescript version, where either an actual promise or an object without a `then` method are required when using async/await.

Reply:
Hmm, I think the fix for this  and other cases should be not using `await` when calling this function instead of adding `as any`, because everything is actually synchronous.

To me the error is here:

```ts
const blah = await page.getDatepicker();
```

Reply:
Yes you are right, made sure there are no cases where it's resolved asynchronously in the e2e tests.